lumicks/pylake/kymotracker/kymoline.py

```diff
@@ -488,3 +489,37 @@ def plot_binding_histogram(self, kind, bins=10, **kwargs):
         plt.bar(edges[:-1], counts, width=widths, align="edge", **kwargs)
         plt.ylabel("Counts")
         plt.xlabel(f"Position ({self[0]._image._position_unit[1]})")
+
+    def _histogram_binding_profile(self, n_time_bins, bandwidth, n_position_points):
+        """Calculate a Kernel Density Estimate (KDE) of binding density along the tether for time bins.
+
+        First the kymograph is binned along the temporal axis. In the case of non-integer `frames / bins`,
+        remaining frames are discarded. Next, for each time bin, a KDE is calculated along the spatial axis.
+
+        Parameters
+        ----------
+        n_time_bins : int
+            Requested number of time bins.
+        bandwidth : float
+            KDE bandwidth; units are in the physical spatial units of the kymograph.
+        n_position_points : int
+            Length of the returned density array(s).
+        """
+
+        n_rows, n_frames = self[0]._image.data.shape
+        position_max = n_rows * self[0]._image._pixel_size
+        bin_size = n_frames // n_time_bins
+        bin_edges = np.arange(n_frames, step=bin_size)
+
+        frames = np.hstack([line.time_idx for line in self])
+        positions = np.hstack([line.position for line in self])
+        bin_labels = np.digitize(frames, bin_edges, right=False)
+
+        x = np.linspace(0, position_max, n_position_points)[:, np.newaxis]
+        densities = []
+        for bin_index in np.arange(n_time_bins) + 1:
+            binned_positions = positions[bin_labels == bin_index][:, np.newaxis]
+            kde = KernelDensity(kernel="gaussian", bandwidth=bandwidth).fit(binned_positions)
+            densities.append(np.exp(kde.score_samples(x)))
```

Comment:
This fails cryptically when `n_position_points` is zero. Should probably assert that it's bigger than zero (or one?).

Reply:
third good catch (I feel like there should be some kind of baseball comment here), handled in the same `try` block as above and rethrow with a better error message; test added.

We'll want to keep the single bin case where you just histogram the whole kymo (requested feature)